packages/react/src/hooks/useSwap.ts

```diff
@@ -2,16 +2,13 @@ import { useAsyncCallback } from "react-async-hook";
 import { WalletNotConnectedError } from "@solana/wallet-adapter-base";
 import { useWallet } from "@solana/wallet-adapter-react";
 import { useStrataSdks } from "./";
-import { ISwapArgs, ITokenBonding } from "@strata-foundation/spl-token-bonding";
+import { ISwapArgs, ITokenBonding, IExtraInstructionArgs, IPostInstructionArgs } from "@strata-foundation/spl-token-bonding";
 import { InstructionResult } from "@strata-foundation/spl-utils";
 import { BN } from "@project-serum/anchor";
 
 export type SwapArgs = {
-  extraInstructions?: (args: {
-    tokenBonding: ITokenBonding;
-    isBuy: boolean;
-    amount: BN | undefined;
-  }) => Promise<InstructionResult<null>>;
+  extraInstructions?: (args: IExtraInstructionArgs) => Promise<InstructionResult<null>>; // instructions executed before swap instructions
+  postInstructions?: (args: IPostInstructionArgs) => Promise<InstructionResult<null>>; // instructions executed after swap instructions
```

Comment:
preInstructions and postInstructions? And then roll extraInstructions (and deprecate) into pre to not be breaking?